Modules/Core/QuadEdgeMesh/include/itkQuadEdgeMeshFrontIterator.hxx

```diff
@@ -132,12 +133,12 @@ template <typename TMesh, typename TQE>
 typename QuadEdgeMeshFrontBaseIterator<TMesh, TQE>::QEType *
 QuadEdgeMeshFrontBaseIterator<TMesh, TQE>::FindDefaultSeed()
 {
-  if (auto * edge = dynamic_cast<QEType *>(m_Mesh->GetEdge()))
+  if (auto * edge = Experimental::Conversion::Convert<QEType *>(m_Mesh->GetEdge()))
   {
     return edge;
   }
   using QEDual = typename QEType::DualType;
-  if (auto * edge = dynamic_cast<QEDual *>(m_Mesh->GetEdge()))
+  if (auto * edge = Experimental::Conversion::Convert<QEDual *>(m_Mesh->GetEdge()))
   {
     return edge->GetRot();
   }
```

Comment:
Now that we use C++17, I think this can be implemented as follows:

```cpp
  auto * edge = m_Mesh->GetEdge(); 

  if constexpr (std::is_convertible_v<decltype(edge), QEType *>)
  {
    return edge;
  }
  using QEDual = typename QEType::DualType;
  if constexpr (std::is_convertible_v<decltype(edge), QEDual *>)
  {
    return edge->GetRot();
  }
```

This would remove the old `dynamic_cast` _and_ make the `if` statements entirely compile-time.